File: mvc1/_etc/Java/9095 1,2,3 더하기.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      int[] dp = new int[12];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 2;
      dp[3] = 4;
      
      for(int i = 4; i < 12; i++) {
          dp[i] = dp[i-1] + dp[i-2] + dp[i-3];
      }
      
      for(int i = 0; i < n; i++) {
          
          int num = s.nextInt();
          System.out.println(dp[num]);
          
      }
      
    }
}

// http://boj.kr/2dc063c00d4c43f8a8093ec5740492fa

File: Beakjoon/_etc/Kotlin/9095 1,2,3 더하기.kt
import java.util.Scanner
import kotlin.math.*

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val num = sc.nextInt()
    
    var dp = Array(12, {0})
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 2
    dp[3] = 4
    
    for(i in 4..11) {
        dp[i] = dp[i-1] + dp[i-2] + dp[i-3]
    }
    
    for(i in 0 until num) {
        var n = sc.nextInt()
        println("${dp[n]}")
    }
}

// http://boj.kr/e541cf715e304809b1b1a208b2605db4
